### PR TITLE
[FS] Ensure all public CCTS methods use locks

### PR DESCRIPTION
**Pull request by quantumagi**

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs [hunk 1]
```diff
@@ -997,37 +1000,37 @@ private OutPoint EarliestOutput(Transaction transaction)
...
-            {
-                this.Synchronize();
-
-                var depositIds = new HashSet<uint256>();
-                foreach (CrossChainTransferStatus status in statuses)
-                    depositIds.UnionWith(this.depositsIdsByStatus[status]);
-
-                uint256[] partialTransferHashes = depositIds.ToArray();
-                ICrossChainTransfer[] partialTransfers = this.Get(partialTransferHashes).Where(t => t != null).ToArray();
+            var depositIds = new HashSet<uint256>();
```
Comment (dangershony) [inline src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs]: Can you be consistent and end the method name with Locked `GetTransfersByStatusInternalLocked` which means the method must be called form a method that has the lock.

**Comment by dangershony** [comment]
Looks good  (apart for my comment)

File: src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs]: The suffix is redundant as all private methods are either called by a public method with a lock or by another private method.

Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs]: Done.

**Pull request merged**